### PR TITLE
api: Implement DELETE /users/{user_id} api endpoint

### PR DESCRIPTION
**Pull request by akashaviator**
This implements DELETE /users/{user_id} api endpoint.

### CODE REVIEW

File: zulip/zulip/__init__.py [hunk 1]
```diff
@@ -1071,6 +1071,20 @@ def get_user_by_id(self, user_id, request=None):
             request=request,
         )
 
+    def deactivate_user_by_id(self, user_id):
+        # type: (int) -> Dict[str, Any]
+
+        '''
+            Example usage:
+
+            >>> client.delete_user_by_id(8)
```
Comment (rht) [inline zulip/zulip/__init__.py:1080]: Why is this called `delete_user_by_id` when the function name is `deactivate_user_by_id`?

Comment (akashaviator) [reply zulip/zulip/__init__.py:1080]: Yeah, that's a typo. I'll fix that.

**Comment by akashaviator** [comment]
@rht  Thanks for pointing that out !

**Comment by akashaviator** [comment]
@rht Fixed it. Thanks for pointing that out !

**Pull request merged**